doc/optional-state.rst

```diff
@@ -0,0 +1,74 @@
+Optional per-peer state
+=======================
+
+An Autocrypt-capable client must store some specific state about each
+of its peers.
+
+This document attempts to describe additional optional state that can
+improve the user experience in some corner cases.
+
+Unlike the standard Autocrypt level 0 state management, some of these
+rules depend on a client being able to keep track of whether it has
+seen a given message before or not, and these guidelines may cause
+non-deterministic results depending on the order that messages are
+encountered.
+
+Additional state
+----------------
+
+Optionally, an agent MAY store and maintain the following data in
```

Comment:
Suggestion: An agent MAY store additional per-peer metadata about observed Autocrypt messages. This can be used to provide more helpful information when user intervention is required.